src/aspire/source/image.py

```diff
@@ -157,11 +161,31 @@ def __init__(self, L, n, dtype="double", metadata=None, memory=None):
         self.generation_pipeline = Pipeline(xforms=None, memory=memory)
         self._metadata_out = None
 
-        # Instantiate the accessor for the `images` property
-        self._img_accessor = _ImageAccessor(self._images, self.n)
-
         logger.info(f"Creating {self.__class__.__name__} with {len(self)} images.")
 
+    @property
+    def n(self):
+        return self._n
+
+    @n.setter
+    def n(self, n):
+        """
+        Sets max image index `n` in `src` and associated
+        `ImageAccessor`.
+
+        Used in sources that potentially reduce the set of images.
+
+        :param n: Number of images.
+        """
+
+        # Enforce type, just-in-case.
+        if n != int(n):
+            raise TypeError("`n` must be an integer")
+        n = int(n)
```

Comment:
Do we need this with the type check above?

Reply:
only if I want `n` to be an integer type (and I do).

The check only ensures that we represent an int.

```
>>> n = 1.0
>>> int(n) == n
True
>>> n
1.0
>>> isinstance(n, int)
False
>>> L = [1,2,3]
>>> L[n]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: list indices must be integers or slices, not float
>>>
```

we are protecting against when n is not actually an  `int`.  The conversion is silent, so we don't know something bad was passed in.

```
>>> n = 1.5
>>> int(n) == n
False
>>> int(n)
1
>>>
```

Reply:
Ah, thanks for the explanation!